### PR TITLE
Add nitrogen mineralization feature with NMI API integration

### LINKED ISSUE #539: Mineralisatie (Nitrogen Mineralization) App for FDM
## Summary

Add a **Mineralisatie** module to the FDM app that visualizes nitrogen mineralization dynamics per field and farm. The feature uses the NMI API endpoints (`/bemestingsplan/nsupply` and `/bemestingsplan/dyna`) to calculate daily cumulative N mineralization curves and dynamic nitrogen advice. ~95% of required input data is already in FDM.

## Motivation

- **Farmer value**: Understanding when and how much nitrogen the soil naturally releases enables farmers to optimize fertilizer timing, reduce costs, and stay within Dutch nitrogen regulations (2026 framework).
- **Low friction**: Soil analysis data, field geometry, cultivation data, and fertilizer applications are already entered by users — only weather data is external (and the NMI API supplies it automatically).
- **Ecosystem synergy**: Complements the existing Nitrogen Balance, Nutrient Advice, and Norms features to create a complete nitrogen management suite.
- **Differentiation**: No other Dutch farm management tool offers real-time, field-specific mineralization curves combined with dynamic fertilizer advice.

### Phase 1 — Core Mineralization (MVP)

**Farm overview** (`/farm/{id}/{cal}/mineralisatie`):
- KPI cards: weighted-average farm N supply, per-field average, highest-supplying field
- Farm-average mineralization curve (area chart, cumulative kg N/ha over the year)
- Sortable field list with annual N supply values and data completeness status
- Method selector dropdown: MINIP, PMN, or Century model

**Field detail** (`/farm/{id}/{cal}/mineralisatie/{b_id}`):
- KPI cards: N supply, active method, soil type, organic matter %
- Multi-method comparison chart: all 3 mineralization methods overlaid on one area chart
- "Today" marker showing current mineralization progress
- Soil data summary table with source and date for each parameter
- Data completeness indicator with warnings for missing/estimated parameters
- Generated Dutch-language insights (compare to farm average, season progress, recommendations)

### Phase 2 — Dynamic N Advice (DYNA)

**DYNA simulation** (`/farm/{id}/{cal}/mineralisatie/{b_id}/dyna`):
- N availability vs. crop uptake dual-line chart with min-max scenario band
- Fertilizer application markers on the chart
- Nitrogen balance summary (artificial, organic, green manure, preceding crop)
- Fertilizer timing and dose recommendation card
- Harvest timing recommendation
- Nitrate leaching cumulative curve

### Phase 3 — Enhanced Features

- PDF/CSV export of mineralization reports
- Historical year-over-year comparison
- Cross-links to Nitrogen Balance and Nutrient Advice features
- Optional weather data integration for improved accuracy

### Integration

- New server-side integration module: `app/integrations/mineralisatie.server.ts`
- Calls NMI API `POST /bemestingsplan/nsupply` (Phase 1) and `POST /bemestingsplan/dyna` (Phase 2)
- Reuses existing NMI API key from `serverConfig.integrations.nmi.api_key`
- Response validation with Zod schemas
- In-memory caching: 24h TTL for nsupply, 24h TTL for dyna

### Data Mapping

All required soil parameters are available from `getCurrentSoilData()` via fdm-core:
`a_som_loi`, `a_clay_mi`, `a_silt_mi`, `a_sand_mi`, `a_c_of`, `a_cn_fr`, `a_n_rt`, `a_n_pmn`, `b_soiltype_agr`

Field coordinates from `field.b_centroid`, crop codes from `cultivations.b_lu_catalogue` (strip `nl_` prefix → BRP code).

Weather data is **not supplied** — the NMI API automatically uses Dutch historical measurements for past days and climatic averages for future days.

### Routes

Mineralisatie is a standalone app (not under Balans) following `farm.$b_id_farm.$calendar.mineralisatie.*`:

```
app/routes/
  farm.$b_id_farm.$calendar.mineralisatie.tsx          → Layout
  farm.$b_id_farm.$calendar.mineralisatie._index.tsx   → Farm overview
  farm.$b_id_farm.$calendar.mineralisatie.$b_id.tsx    → Field detail
  farm.$b_id_farm.$calendar.mineralisatie.$b_id.dyna.tsx → DYNA (Phase 2)
```

### UI Components

New components under `app/components/blocks/mineralisatie/`:
- `mineralisatie-chart.tsx` — recharts AreaChart for cumulative N supply curves
- `nsupply-kpi.tsx` — summary KPI cards
- `data-completeness.tsx` — parameter status indicator
- `method-selector.tsx` — MINIP/PMN/Century dropdown
- `field-list.tsx` — sortable field table
- `skeletons.tsx` — Suspense loading fallbacks
- Phase 2: `dyna-chart.tsx`, `dyna-balance.tsx`, `dyna-advice.tsx`, `leaching-chart.tsx`

### Sidebar

Add "Mineralisatie" as a new standalone app entry in the **Apps** section of the sidebar (same level as Balans), with a `Thermometer` icon.

### Phase 1 — Core Mineralization (MVP)

- [ ] Create `mineralisatie.server.ts` integration module (`getNSupplyForField`, `getNSupplyForFarm`, `buildNSupplyRequest`, `assessDataCompleteness`, Zod schemas)
- [ ] Create layout route with breadcrumb and field selector
- [ ] Create farm overview route + loader with async streaming
- [ ] Create field detail route + loader with 3-method parallel fetch
- [ ] Create `mineralisatie-chart.tsx` (recharts AreaChart, multi-method overlay, today marker)
- [ ] Create `nsupply-kpi.tsx` KPI cards
- [ ] Create `data-completeness.tsx` component
- [ ] Create `method-selector.tsx` dropdown
- [ ] Create `field-list.tsx` sortable field table
- [ ] Create `skeletons.tsx` Suspense fallbacks
- [ ] Add sidebar link as standalone app in Apps section
- [ ] Write integration tests for API mapping and completeness logic

### Phase 2 — Dynamic N Advice

- [ ] Add `getDynaForField`, `buildDynaRequest` to integration module
- [ ] Create DYNA route + loader
- [ ] Create `dyna-chart.tsx` (N availability vs. uptake with min-max band)
- [ ] Create `dyna-balance.tsx` (nitrogen balance card)
- [ ] Create `dyna-advice.tsx` (fertilizer + harvest recommendation card)
- [ ] Create `leaching-chart.tsx` (nitrate leaching curve)
- [ ] Add DYNA CTA banner to field detail page

### Phase 3 — Enhanced Features

- [ ] Farm aggregate weighted-average calculations
- [ ] PDF/CSV export
- [ ] Historical year-over-year comparison
- [ ] Cross-links to N Balance and Nutrient Advice
- [ ] Optional weather data integration

## Data Completeness

The feature includes a data completeness system that:
- Evaluates which soil parameters are available vs. missing for the selected method
- Calculates a completeness score (0–100%)
- Shows inline warnings for missing/estimated parameters
- Links to soil analysis upload when data is incomplete
- NMI API can still run with partial data (uses estimates) but results may be less accurate

## Error Handling

- **422 (missing data)**: show data completeness indicator with specific missing fields
- **503 (API unavailable)**: show cached result or friendly message
- **Per-field isolation**: one failing field does not break the farm overview
- **Graceful degradation**: fields without soil analysis show ⚠️ status, not errors

## Risks

| Risk | Likelihood | Mitigation |
|------|------------|------------|
| NMI API endpoints are experimental | Medium | Version-pin calls, robust error handling, monitor changelog |
| Missing soil data reduces accuracy | Medium | Data completeness indicators, NMI estimates fallback |
| Many fields slow down farm overview | Low | Parallel fetching, 24h cache, consider pagination |
| BRP code mapping edge cases | Low | Validate `nl_` prefix parsing in tests |

### PR DESCRIPTION
**Pull request by SvenVw**
## Summary by CodeRabbit

* **New Features**
  * Added Mineralisatie app: farm & field pages with nitrogen supply curves, DYNA dynamics, leaching charts, KPIs, and per-field results.
  * Field picker, method selector, and per-field DYNA page with advice and nitrogen-balance cards.
  * Data completeness card with score, parameter lists, and CTA to add soil analyses; animated skeleton fallbacks.

* **Navigation & UI**
  * Sidebar entry, responsive breadcrumb header and links for Mineralisatie and DYNA.

* **Tests & Chores**
  * New calculator tests and release metadata updates.

Closes #539

### CODE REVIEW

File: fdm-calculator/src/mineralization/assessment.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            soilData[p] !== null &&
+            soilData[p] !== undefined &&
+            soilMeta?.[p]?.source !== NMI_SOURCE,
+    ).length
+    const availableOptional = optional.filter(
+        (p) =>
+            soilData[p] !== null &&
+            soilData[p] !== undefined &&
+            soilMeta?.[p]?.source !== NMI_SOURCE,
+    ).length
```
[the remark this answers is not in the document]
Comment (SvenVw) [reply fdm-calculator/src/mineralization/assessment.ts:138]: Estimates from NMI BodemSchat should be used in case no soil analyis is present, but not count towards available soil parameters

**Comment by SvenVw** [comment]
@coderabbitai resume

**Review by BoraIneviNMI — APPROVED** [review]
I don't see anything particular. The cache seems to work properly and the pages are responsive.

**I would change the text of the NSupplyMethod selector on the main mineralization route. It shows an ellipsis because of the description part of the dropdown.** It is easy to add your own text but I am trying to find a way to do it with <SelectValue />
